pureport/api/client.py

```diff
@@ -860,10 +871,16 @@ def update(self, connection, wait_until_active=False):
             :rtype: Connection
             :raises: .exception.HttpClientException
             :raises: .exception.ConnectionOperationTimeoutException
+            :raises: .exception.ConnectionOperationFailedException
             """
             connection = self.__session.put(connection['href'], json=connection).json()
             if wait_until_active:
-                return Client.ConnectionsClient.get_connection_until_state(self.__session, connection, 'ACTIVE')
+                return Client.ConnectionsClient.get_connection_until_state(
+                    self.__session,
+                    connection,
+                    'ACTIVE',
+                    'FAILED_TO_UPDATE'
```

Comment:
It might be useful to have a list of all possible states somewhere and reference those instead of these strings.

Reply:
Why not pass in a list here instead of variable parameters?